metrics/src/metrics.rs

```diff
@@ -159,25 +159,13 @@ impl MetricsAgent {
         Self { sender }
     }
 
-    fn collect_points(
-        points_map: &mut HashMap<log::Level, (CounterMap, Vec<DataPoint>)>,
-    ) -> Vec<DataPoint> {
-        let points: Vec<DataPoint> = [
-            Level::Error,
-            Level::Warn,
-            Level::Info,
-            Level::Debug,
-            Level::Trace,
-        ]
-        .iter()
-        .filter_map(|level| points_map.remove(level))
-        .flat_map(|(counters, points)| {
-            let counter_points = counters.into_iter().map(|(_, v)| v.into());
-            points.into_iter().chain(counter_points)
-        })
-        .collect();
-        points_map.clear();
-        points
+    fn collect_points(points: &mut Vec<DataPoint>, counters: &mut CounterMap) -> Vec<DataPoint> {
+        let mut ret: Vec<DataPoint> = Vec::default();
+        std::mem::swap(&mut ret, points);
+        for (_, v) in counters.drain() {
+            ret.push(v.into());
```

Comment:
Do you know how large 'counters' can be? It may be worth reserving the space in ret for adding all of counters before pushing.